### ISSUE TITLE
Static typing of metric values

### ISSUE THREAD

**Issue by streadway** [prometheus/prometheus#185]
https://github.com/prometheus/prometheus/blob/master/retrieval/format/processor0_0_1.go#L56

Is a context sensitive union type that can produce runtime errors when the "type" key does not match the expected type of the value of the "value" key.  Consider using a key name for the type rather than a "type" key with the metric type as a value.  Like:

```json
[
  {"metric": {"labels":[], "count": 1} },
  {"metric": {"labels":[], "histogram": { "0.01": 15, "0.1": 20 }
]
``` 

This means that collectors ignore metrics they don't know how to join.  If you want to be explicit, keep the "type" key as the tag for the union of possible metric types.  Then the schema can use static typing for possible metric values.

TL;DR: Do not use interface{} types inside in the schema definition for significant fields.

**Comment by fabxc** [comment]
JSON format 0.0.1, I think we are over it.
